### DISCUSSION TITLE
Database migration crashes during fresh installation (docker-compose)

### DISCUSSION THREAD

**Discussion by la9vka** [wavelog/wavelog#781] [Q&A] [answered]
I'm running the original docker-compose, no proxy (as timeouts was related to a similar discussion) or any changes. But after a few seconds, it fails on Running database migrations.

The installer_debug.log says:
```
Wavelog Installer Debug Log
-------
Log Location: ../application/logs/installer_debug.log

DEBUG - 2024-08-20 12:09:07 --> Start writing config.php
DEBUG - 2024-08-20 12:09:08 --> File: config.php successfully written
DEBUG - 2024-08-20 12:09:08 --> Start writing database.php
DEBUG - 2024-08-20 12:09:09 --> File: database.php successfully written
DEBUG - 2024-08-20 12:09:09 --> Start creating database structure with assets/install.sql
DEBUG - 2024-08-20 12:09:09 --> Tables successfully created
DEBUG - 2024-08-20 12:09:09 --> Start migrating database to the newest version.
ERROR - 2024-08-20 12:09:16 --> Could not migrate database. Ajax crashed.
```

And the log .. date.. .php file

```
<?php defined('BASEPATH') OR exit('No direct script access allowed'); ?>

DEBUG - 2024-08-20 12:09:09 --> Created migration lockfile
DEBUG - 2024-08-20 12:09:09 --> Migrating up from version 15 to version 016
DEBUG - 2024-08-20 12:09:09 --> Migrating up from version 016 to version 017
DEBUG - 2024-08-20 12:09:09 --> Migrating up from version 017 to version 018
DEBUG - 2024-08-20 12:09:09 --> Migrating up from version 018 to version 019
... lots of subsequent lines of version numbers ...
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 193 to version 194
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 194 to version 195
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 195 to version 196
DEBUG - 2024-08-20 12:09:15 --> Migration class already loaded. Second attempt ignored.
DEBUG - 2024-08-20 12:09:15 --> There is a lockfile for migrations. Checking the age...
DEBUG - 2024-08-20 12:09:15 --> Migration lockfile lifetime in seconds: 6/300
DEBUG - 2024-08-20 12:09:15 --> Migration process is currently locked. Second migration attempt ignored.
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 196 to version 197
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 197 to version 198
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 198 to version 199
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 199 to version 200
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 200 to version 201
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 201 to version 202
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 202 to version 203
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 203 to version 204
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 204 to version 205
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 205 to version 206
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 206 to version 207
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 207 to version 208
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 208 to version 209
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 209 to version 210
ERROR - 2024-08-20 12:09:15 --> Mig: Error at Mig 210 for txt files. Function exec() not usable. Run manually a git reset.
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 210 to version 211
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 211 to version 212
DEBUG - 2024-08-20 12:09:15 --> Migrating up from version 212 to version 213
DEBUG - 2024-08-20 12:09:15 --> Finished migrating to 213
DEBUG - 2024-08-20 12:09:15 --> Deleted migration lockfile
DEBUG - 2024-08-20 12:09:16 --> Migration class already loaded. Second attempt ignored.
```

Deleting volumes and re pulling does not fix it. Any ideas?

**Comment by HB9HIL** [comment]
Hello @la9vka 

Can you confirm, that your `docker-compose.yml` does exactly look like this? (Please compare in detail)

```yaml
---
services:
  wavelog-db:
    image: mariadb:11.3
    container_name: wavelog-db
    environment:
      MARIADB_RANDOM_ROOT_PASSWORD: yes
      MARIADB_DATABASE: wavelog
      MARIADB_USER: wavelog
      MARIADB_PASSWORD: wavelog # <- Insert a strong password here
    volumes:
      - wavelog-dbdata:/var/lib/mysql
    restart: unless-stopped

  wavelog-main:
    container_name: wavelog-main
    image: ghcr.io/wavelog/wavelog:latest
    depends_on:
      - wavelog-db
    environment:
      CI_ENV: docker
    volumes:
      - wavelog-config:/var/www/html/application/config/docker
      - wavelog-uploads:/var/www/html/uploads
      - wavelog-userdata:/var/www/html/userdata
    ports:
      - "8086:80"
    restart: unless-stopped

volumes:
  wavelog-dbdata:
  wavelog-uploads:
  wavelog-userdata:
  wavelog-config:
```

You also find this in our wiki [here](https://github.com/wavelog/wavelog/wiki/Installation-via-Docker).

**Reply by la9vka** [reply]
Yes, except for the added log directory (copied from the wiki). Platform is ubuntu x64 with about 32gb free mem and mostly idle.

```
---
services:
  wavelog-db:
    image: mariadb:11.3
    container_name: wavelog-db
    environment:
      MARIADB_RANDOM_ROOT_PASSWORD: yes
      MARIADB_DATABASE: wavelog
      MARIADB_USER: wavelog
      MARIADB_PASSWORD: wavelog
    volumes:
      - wavelog-dbdata:/var/lib/mysql
    restart: unless-stopped

  wavelog-main:
    container_name: wavelog-main
    image: ghcr.io/wavelog/wavelog:latest
    depends_on:
      - wavelog-db
    environment:
      CI_ENV: docker
    volumes:
      - wavelog-config:/var/www/html/application/config/docker
      - wavelog-uploads:/var/www/html/uploads
      - wavelog-userdata:/var/www/html/userdata
      - /dockerdata/wavelog/logs:/var/www/html/application/logs
    ports:
      - "8086:80"
    restart: unless-stopped

volumes:
  wavelog-dbdata:
  wavelog-uploads:
  wavelog-userdata:
  wavelog-config:
```

**Reply by HB9HIL** [reply]
Looks like we have to seperate issues here. The Error in Mig 210 can be ignored and has nothing to do with the crashed ajax. Which browser do you use?

**Reply by la9vka** [reply, marked as answer]
Hmm.. I started in Firefox, but when i changed to Chrome I noticed that my configuration was present. I redid the test, wiping the volumes and images (not just selecting repull). And it completed in Chrome. Wiped again and now it worked with Firefox also. Not sure why it failed the first time, but thank you for the help :)

**Reply by HB9HIL** [reply]
Strange things happen. But in fact I can't reproduce your issue 😂👍 The Error in Mig 210 can be ignored (I will fix that)

**Reply by erangaj** [reply]
I copied and pasted the exact YAML to my Synology container station (Docker) and got the same issue again. Finally, I created a VM an manually installed it on LAMP stack. It went smoothly and I am using it now. Before that I attempted to install on docker on 2 different hosts, but failed each time.

**Comment by HB9HIL** [comment]
For documentation: Fixed the error message in migration 210 with commit https://github.com/wavelog/wavelog/commit/16cf6e766ff83faf3cb49285113517dba12a6c9b

**Comment by OH6OKSA** [comment]
I try to do the installation in Unraid 6.12.13 running docker version 24.0.9 and I always end up with same problem discussed here earlier. Deleted volumes, re-installed and etc...no help always same output as below?

                            Wavelog Installer Debug Log
-------
Log Location: ../application/logs/installer_debug.log

DEBUG - 2024-10-17 10:32:22 --> Start writing config.php
DEBUG - 2024-10-17 10:32:23 --> File: config.php successfully written
DEBUG - 2024-10-17 10:32:23 --> Start writing database.php
DEBUG - 2024-10-17 10:32:24 --> File: database.php successfully written
DEBUG - 2024-10-17 10:32:24 --> Start creating database structure with assets/install.sql
DEBUG - 2024-10-17 10:32:42 --> Tables successfully created
DEBUG - 2024-10-17 10:32:42 --> Start migrating database to the newest version.
ERROR - 2024-10-17 10:37:07 --> Could not migrate database. Ajax crashed.

**Reply by HB9HIL** [reply]
do you use a reverse proxy to reach the container?

**Reply by OH6OKSA** [reply]
No I do not, but seems to related to the password change. When ever I changed default password: wavelog to something else it did not work. With defaulf docker-compose.yml it is working with out any change. I repeated this several times with clean install with same results...hard to say what is the problem...but with deafults it up and running :-)

MARIADB_PASSWORD: wavelog # <- Insert a strong password here

**Reply by HB9HIL** [reply]
does the new password contain any special characters?

**Reply by HB9HIL** [reply]
please tell me one example password you used, which did not work so I can try to track this down

**Reply by OH6OKSA** [reply]
Like this type password (format same not the actual password) June2024!

**Comment by erangaj** [comment]
I am having the same issue.The log is identical to the above. I get an 404 error when running migrate script.

The database password is text only. (no special chars.) 

<img width="976" alt="Untitled" src="https://github.com/user-attachments/assets/53e73beb-44db-4614-bc59-8d4512937649">

<img width="725" alt="Untitled 2" src="https://github.com/user-attachments/assets/68e967f8-65c1-4b17-8173-a61641fc769f">


Any thoughts?

**Comment by HB9HIL** [comment]
@erangaj The issue you experienced (migration crashed) has nothing to do with the database password. The installation would not be possible if the installer does not have any connection to the database (you would stuck at the fourth tab)

calling the migration is called over the webserver. To test properly I'd need much more information. Docker or Plain Install? If Plain, which webserver and config, and so on.. Pretty difficult to say whats wrong and even more difficult to fix

**Reply by erangaj** [reply]
It is a Docker install. I don't use a reverese proxy.
